### PR TITLE
Added ecs-anywhere role and anisble-galaxy collection files

### PR DESCRIPTION
**Pull request by sergio-gaggle**
Added ecs-anywhere role and tasks files. Also added manifest file for ansible-galaxy collection packaging.

### CODE REVIEW

File: MANIFEST.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  "authors": [
+   "Sergio Mejia <sergio@gaggle.net>"
+  ],
+  "readme": "README.md",
+  "tags": [
+   "aws", "ecs", "centos"
+  ],
+  "description": "ECS Anywhere Collection",
+  "license": [],
+  "license_file": "LICENSE",
```
Comment (cplee) [inline MANIFEST.json:14]: `LICENSE` file is missing - please add an MIT license file to the repo

File: galaxy.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+### OPTIONAL but strongly recommended
+
+# A short summary description of the collection
+description: AWS ECS Anywhere Collection
+
+# Either a single license or a list of licenses for content inside of a collection. Ansible Galaxy currently only
+# accepts L(SPDX,https://spdx.org/licenses/) licenses. This key is mutually exclusive with 'license_file'
+license:
+- GPL-2.0-or-later
```
Comment (cplee) [inline galaxy.yml]: no GPL please...let's use MIT

File: roles/ecs_anywhere/meta/main.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+---
+galaxy_info:
+  role_name: ecs-anywhere
+  author: gagglenet
+  descpription: Ansible role that configures ecs anywhere instance
+  company: Gaggle Net
+  issue_tracker_url: https://github.com/gaggle-net/ansible-ecs-anywhere-playbook/issues
+  license: Apache-2.0
```
Comment (cplee) [inline roles/ecs_anywhere/meta/main.yml]: change to MIT license for consistency

File: README.md [hunk 4]
```diff
@@ -1 +1,44 @@
...
+```yaml
+---
+- hosts: aws_ecs
+  tasks:
+    - import_role:
+        name: gagglenet.ecs_anywhere.ecs_anywhere
+  become: yes
+```
+
+## Running the Playbook
```
Comment (cplee) [inline README.md:49]: Would be good to document the parameters that are `required` and `optional`

File: roles/ecs_anywhere/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    path: "{{ aws_dir }}"
+    state: directory
+    owner: root
+    group: root
+    mode: 0611
+  become: yes
+
+- name: ECS Anywhere | Download ECS Anywhere Install Script
+  get_url: 
+    url: https://amazon-ecs-agent-packages-preview.s3.us-east-1.amazonaws.com/ecs-anywhere-install.sh
```
Comment (cplee) [inline roles/ecs_anywhere/tasks/main.yml]: the bucket name should come from `aws_bucket_name` parameter. Also, shouldn't use `preview` anymore as per: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-anywhere-registration.html

File: roles/ecs_anywhere/defaults/main.yml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+---
+aws_region: us-west-2
+aws_dir: /opt/amazon/ecs-anywhere
+aws_ecs_cluster: default
+aws_s3_bucket: amazon-ecs-agent-packages-preview
```
Comment (cplee) [inline roles/ecs_anywhere/defaults/main.yml]: shouldn't use `preview` anymore as per: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-anywhere-registration.html

File: roles/ecs_anywhere/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    dest: "{{ aws_dir }}"
+    mode: '0550'
+    checksum: sha256:https://amazon-ecs-agent-packages-preview.s3.us-east-1.amazonaws.com/ecs-anywhere-install.sh.sha256
+  become: yes
+
+- name: ECS Anywhere | Create SSM Activation Code
+  local_action: ansible.builtin.command /usr/bin/aws ssm create-activation --iam-role {{ aws_ssm_role }} --registration-limit {{ ansible_play_hosts | length }} --profile {{ aws_profile }}
```
Comment (cplee) [inline roles/ecs_anywhere/tasks/main.yml]: `create-activation` is called in 3 separate places - can this be simplified?

File: roles/ecs_anywhere/tasks/register-ssm-instance.yml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    name: ecs.service
+    state: stopped
+
+- name: ECS Anywhere | Delete ECS Agent Data
+  file:
+    path: /var/lib/ecs/data/agent.db
+    state: absent
+
+- name: ECS Anywhere | Create SSM Activation Code
+  local_action: ansible.builtin.command /usr/bin/aws ssm create-activation --iam-role {{ aws_ssm_role }} --registration-limit {{ ansible_play_hosts | length }} --profile {{ aws_profile }}
```
Comment (cplee) [inline roles/ecs_anywhere/tasks/register-ssm-instance.yml]: `create-activation` is called in 3 separate places - can this be simplified?

File: roles/ecs_anywhere/tasks/aws-script-install.yml [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+- name: Download ECS Anywhere Install Script
+  get_url: 
+    url: https://amazon-ecs-agent-packages-preview.s3.us-east-1.amazonaws.com/ecs-anywhere-install.sh
+    dest: "{{ aws_dir }}"
+    mode: '0550'
+    checksum: sha256:https://amazon-ecs-agent-packages-preview.s3.us-east-1.amazonaws.com/ecs-anywhere-install.sh.sha256
+  become: yes
+
+- name: create ssm activation code
+  local_action: ansible.builtin.command /usr/bin/aws ssm create-activation --iam-role {{ ssm_role }} --registration-limit {{ ansible_play_hosts | length }} --profile {{ aws_profile }}
```
Comment (cplee) [inline roles/ecs_anywhere/tasks/aws-script-install.yml]: `create-activation` is called in 3 separate places - can this be simplified?

File: roles/ecs_anywhere/tasks/aws-script-install.yml [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    group: root
+    mode: 0611
+  become: yes
+
+- name: Download ECS Anywhere Install Script
+  get_url: 
+    url: https://amazon-ecs-agent-packages-preview.s3.us-east-1.amazonaws.com/ecs-anywhere-install.sh
+    dest: "{{ aws_dir }}"
+    mode: '0550'
+    checksum: sha256:https://amazon-ecs-agent-packages-preview.s3.us-east-1.amazonaws.com/ecs-anywhere-install.sh.sha256
```
Comment (cplee) [inline roles/ecs_anywhere/tasks/aws-script-install.yml]: these are the old buckets from preview: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-anywhere-registration.html

**Review by cplee — CHANGES_REQUESTED** [review]
Can we make `aws_profile` optional? We use profiles a lot internally, but that isn't the best assumption for others...some may not use profiles the way we do.

Also, there is a lot of dependency on the AWS CLI - any reason why we couldn't use Ansible plugins and write the code in Python using boto3 to remove the need for the AWS CLI?

File: MANIFEST.json (hunk 1 above)
Comment (sergio-gaggle) [reply MANIFEST.json:14]: Added MIT license file and wrote modules to remove dependency on the AWS CLI. aws_ssm_activation module creates/gets/deletes SSM activations and aws_ssm_instance module gets/de-registers instances from SSM

File: galaxy.yml (hunk 2 above)
Comment (sergio-gaggle) [reply galaxy.yml]: Updated to MIT

File: roles/ecs_anywhere/meta/main.yml (hunk 3 above)
Comment (sergio-gaggle) [reply roles/ecs_anywhere/meta/main.yml]: Updated to MIT

**Comment by cplee** [comment]
@sergio-gaggle - this is looking REALLY good 🤩

Only 1 last request before we merge it - can you update the `README.md` to include a list of vars like:

| variable | description | required? | default |
| -------- | ----------- | ---------- | ------- | 
| iam_role | Which IAM role to associate with instance | yes | - |
| aws_ecs_cluster | Name of ECS cluster to use | yes | - |
| aws_region | Which region to use for this ECS instance | no | `us-west-2` |
| aws_install_dir | Where to install ECS anywhere | no | `/opt/amazon/ecs-anywhere` |
| aws_script_url | Location of install script | no | `https://amazon-ecs-agent.s3.amazonaws.com/ecs-anywhere-install-latest.sh` |

**Comment by tmz-gaggle** [comment]
Nice work putting this together Sergio!

Is Fedora really a supported OS for the collection?  ECS Anywhere itself says versions 32 and 33 are supported, which means they're  a month or two away from not supporting any current Fedora releases.  Fedora 32 was EOL in May.  Fedora 33 will be EOL 4 weeks after Fedora 35 is released, which is scheduled for October 19 or 26.  The versions in the collection here are 27 and 28, which have been EOL since November 2018 and May 2019, respectively.

It might be worth removing Fedora entirely unless we want to test it on Fedora 33 and update the README and role metadata?

Not the most useful of reviews, I know.  I just noticed those old versions in the README and thought I'd ask.

File: README.md (hunk 4 above)
Comment (sergio-gaggle) [reply README.md:49]: Added section to README describing which variables are optional or required.

**Review by cplee — APPROVED** [review]

**Comment by cplee** [comment]
@sergio-gaggle - ready to merge?

**Comment by sergio-gaggle** [comment]
@cplee yup, ready to merge to master

**Pull request merged**